src/storage/src/source/source_reader_pipeline.rs

```diff
@@ -721,7 +721,7 @@ where
                                     key: message.key,
                                     value: message.value,
                                     metadata: message.metadata,
-                                    from_time: from_ts,
+                                    from_time: message.from_time,
```

Comment:
and here we preserve the original (not fast forwarded timestamp) for everything that comes after reclocking. This is important for upsert to work correctly as it needs to differentiate between updates that happened at different source times